File: api-java-sql/src/main/java/com/nebulahotel/repository/HospedagensRepository.java
package com.nebulahotel.repository;

import com.nebulahotel.model.Hospedagens;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import java.math.BigDecimal;
import java.util.Optional;

public interface HospedagensRepository extends JpaRepository<Hospedagens, Integer> {

    Optional<Hospedagens> findByIdReserva(Integer idReserva);

    @Query("SELECT COALESCE(SUM(h.totalGeral), 0) FROM Hospedagens h WHERE LOWER(h.status) = 'encerrada'")
    BigDecimal calcularFaturamentoTotal();
}

File: api-kotlin-nosql/src/main/kotlin/com/nebulahotel/repository/HospedagensRepository.kt
package com.nebulahotel.repository

import com.nebulahotel.model.Hospedagens
import org.springframework.data.mongodb.repository.MongoRepository

interface HospedagensRepository : MongoRepository<Hospedagens, Int> {
    fun findByIdReserva(idReserva: Int): Hospedagens?
}
